browser/ntp_sponsored_images/ntp_sponsored_images_service_factory.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_NTP_SPONSORED_IMAGES_NTP_SPONSORED_IMAGES_SERVICE_FACTORY_H_
+#define BRAVE_BROWSER_NTP_SPONSORED_IMAGES_NTP_SPONSORED_IMAGES_SERVICE_FACTORY_H_
+
+#include "base/memory/singleton.h"
+#include "components/keyed_service/content/browser_context_keyed_service_factory.h"
+
+class NTPSponsoredImagesServiceFactory
+    : public BrowserContextKeyedServiceFactory {
+ public:
+  static NTPSponsoredImagesServiceFactory* GetInstance();
+
+  NTPSponsoredImagesServiceFactory(
```

Comment:
`DISALLOW_COPY_AND_ASSIGN(NTPSponsoredImagesServiceFactory)` should be used instead of these

Reply:
I'm pretty sure that is deprecated and raises lint warnings. At least usage of those macros threw errors when contributing upstream on M80

Reply:
From "C++  Do's and Don'ts":
https://chromium.googlesource.com/chromium/src.git/+/master/styleguide/c++/c++-dos-and-donts.md
> Declare or delete these operations in the public section, between other constructors and the destructor; DISALLOW_COPY_AND_ASSIGN is deprecated. For a non-copyable/movable type, delete the copy operations (the move operations will be implicitly deleted); otherwise, declare either copy operations, move operations, or both (a non-declared pair will be implicitly deleted). Always declare or delete both construction and assignment, not just one (which can introduce subtle bugs).

Reply:
so @petemill linked me to some discussion about this in upstream and I can't say I'm particularly moved by the arguments, but I guess I can go either way on it

Reply:
Well, I think starting with M80, `cpplint.py` is going to throw lint errors- so I think we have to use this if we want to continue using the linter